android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java

```diff
@@ -327,15 +295,6 @@ public boolean onEditorAction(TextView focusTextView, int actionId, KeyEvent eve
                     return false;
                 }
             });
-            // generate repeating groups on focus change
-            referenceEditText.setOnFocusChangeListener(new View.OnFocusChangeListener() {
-                @Override
-                public void onFocusChange(View view, boolean hasFocus) {
-                    if (!hasFocus) {
-                        addOnDoneAction((TextView) view, doneButton, widgetArgs);
-                    }
-                }
-            });
```

Comment:
Any reason to remove this? This was recommended as a better user experience by Roger https://github.com/onaio/rdt-standard/issues/319#issuecomment-691212698.

Reply:
Yeah, this feature breaks the normal behaviour of repeating group, it adds a bug whereby it creates twice the number of groups needed.

Reply:
Ok, were you able to figure out why that happens?

Reply:
`referenceEditText` gains focus when cursor is on it, a number is entered ...  then after `referenceEditText` loses focus when the done button is clicked it fires the `addOnDoneAction` from focusChange listener at the same time the done button clickListener does the same ... so it generates twice the number of groups.

Reply:
Cool. Would removing the `addOnDoneAction` from the done button work then?

Reply:
yeah ... thats why i proposed that to use the feature one has to hide the done button.

Reply:
I meant that since onFocusChanged is triggered when the button is clicked (after entering text) we can simply use onFocusChanged and remove the onclicklistener from the done button.

Reply:
I dont think thats a good idea, looks like a hack ... i think one has to be disabled.

Reply:
It really isn't. The point of the feature (from Roger's comment) is to generate the repeating group when one navigates away from the reference edit text. Clicking on other views (including the done button) counts as navigating away.

Reply:
Why would you need the done button if you have a onFocusChangeListener on a referenceEditText? I dont get it but anyways ... This needs to be configurable since not all implementation expect that behaviour.

Reply:
tbh, it's only for backward compatibility, we don't need a button any more. The button being "truly" clickable or not doesn't really change the end-user experience so that's not a problem.

You can make it configurable, but any time someone wants both the button and focus change to be supported, the bug will show up again.